audiomentations/augmentations/room_simulator.py

```diff
@@ -0,0 +1,355 @@
+                self.parameters["source_y"],
+            ),
+        )
+        self.parameters["source_z"] = max(
+            0,
+            min(
+                self.parameters["size_z"] - self.padding,
+                self.parameters["source_z"],
+            ),
+        )
+
+        self.parameters["mic_radius"] = random.uniform(
+            self.min_mic_radius, self.max_mic_radius
+        )
+        self.parameters["mic_azimuth"] = random.uniform(
+            self.min_mic_azimuth, self.max_mic_azimuth
+        )
+        self.parameters["mic_elevation"] = random.uniform(
+            self.min_mic_elevation, self.max_mic_elevation
+        )
+
+        # Convert to cartesian coordinates according to ADM
+        mic_x = self.parameters["source_x"] - self.parameters["mic_radius"] * np.cos(
+            self.parameters["mic_elevation"]
+        ) * np.sin(self.parameters["mic_azimuth"])
+        mic_y = self.parameters["source_y"] + self.parameters["mic_radius"] * np.cos(
+            self.parameters["mic_elevation"]
+        ) * np.cos(self.parameters["mic_azimuth"])
+        mic_z = self.parameters["source_z"] + self.parameters["mic_radius"] * np.sin(
+            self.parameters["mic_elevation"]
+        )
+
+        # Clamp between 0 and room dimensions
+        self.parameters["mic_x"] = max(
+            0, min(self.parameters["size_x"] - self.padding, mic_x)
+        )
+        self.parameters["mic_y"] = max(
+            0, min(self.parameters["size_y"] - self.padding, mic_y)
+        )
+        self.parameters["mic_z"] = max(
+            0, min(self.parameters["size_z"] - self.padding, mic_z)
+        )
+
+    def apply(self, samples, sample_rate):
+        try:
+            import pyroomacoustics as pra
+        except ImportError:
+
+            print(
+                "Failed to import pyroomacoustics. Maybe it is not installed? "
+                "To install the optional pyroomacoustics dependency of audiomentations,"
+                " do `pip install audiomentations[extras]` or simply "
+                " `pip install pyroomacoustics`",
+                file=sys.stderr,
+            )
+            raise
+
+        assert samples.dtype == np.float32
+
+        # It makes no sense to use stereo data.
+        if len(samples.shape) > 1:
+            samples = samples.mean(0)
+
+        # Construct room
+        self.room = pra.Room.from_corners(
+            np.array(
+                [
+                    [0, 0],
+                    [0, self.parameters["size_x"]],
+                    [self.parameters["size_x"], self.parameters["size_y"]],
+                    [self.parameters["size_y"], 0],
+                ]
+            ).T,
+            fs=sample_rate,
+            materials=pra.Material(self.parameters["absorption_coefficient"]),
+            ray_tracing=self.use_ray_tracing,
+            air_absorption=True,
+        )
+
+        if self.use_ray_tracing:
+            # TODO: Somehow make those parameters
```

Comment:
Is this something you intend to address in a future PR?

Reply:
I might add on this one. Don't know how useful those are.